integration/spark/shared/src/main/java/io/openlineage/spark/agent/lifecycle/plan/KustoRelationVisitor.java

```diff
@@ -0,0 +1,161 @@
+package io.openlineage.spark.agent.lifecycle.plan;
+
+import io.openlineage.client.OpenLineage;
+import io.openlineage.spark.api.DatasetFactory;
+import io.openlineage.spark.api.OpenLineageContext;
+import io.openlineage.spark.api.QueryPlanVisitor;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.reflect.FieldUtils;
+import org.apache.spark.api.java.Optional;
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan;
+import org.apache.spark.sql.execution.datasources.LogicalRelation;
+import org.apache.spark.sql.sources.BaseRelation;
+import org.apache.spark.sql.sources.CreatableRelationProvider;
+import org.apache.spark.sql.types.StructType;
+
+/** */
+@Slf4j
+public class KustoRelationVisitor<D extends OpenLineage.Dataset>
+    extends QueryPlanVisitor<LogicalRelation, D> {
+
+  private final DatasetFactory<D> factory;
+  private static final String KUSTO_CLASS_NAME =
+      "com.microsoft.kusto.spark.datasource.KustoRelation";
+
+  private static final String KUSTO_PROVIDER_CLASS_NAME =
+      "com.microsoft.kusto.spark.datasource.DefaultSource"; // Not used
+
+  private static final String KUSTO_URL_SUFFIX = ".kusto.windows.net";
+
+  private static final String KUSTO_PREFIX = "azurekusto://";
+
+  public KustoRelationVisitor(OpenLineageContext context, DatasetFactory<D> factory) {
+    super(context);
+    this.factory = factory;
+  }
+
+  public static boolean isKustoClass(LogicalPlan plan) {
+    return plan instanceof LogicalRelation
+        && ((LogicalRelation) plan).relation().getClass().getName().equals(KUSTO_CLASS_NAME);
```

Comment:
You might consider loading the class (as you do in `hasKustoClasses`), then using `Class.isAssignableFrom()` - this allows you to support subclasses in the future.

Reply:
One quick question on this point @collado-mike - if we want to go this route, we would need to first import the class right? Would we need to use compileOnly for that?

Reply:
@collado-mike Are you saying  we will support subclassing as in if someone else has created a subclass of the Kusto spark connector?  While it's possible, I'm less confident someone would actually take the time to do so given it's a product and not entirely an open source compute engine.  That's not to say we don't want to discourage Kusto users from writing whatever code they need to but I've never seen a company modify / extend one of Microsoft's pre-built SDKs.

Reply:
@hmoazam, no you don't need to import it - you can load the class as you do in `hasKustoClasses`, ie.,
```
Class c = getClass().getClassLoader()
          .loadClass("com.microsoft.kusto.spark.datasource.DefaultSource")
```
then use the `isAssignableFrom` method on the class object
```
return c.isAssignableFrom(plan.relation().getClass());
```

@wjohnson , I was actually thinking if the Kusto library itself ends up using subclasses for whatever reason. It's not uncommon for a library to end up using subclasses as part of a refactoring or adding new functionality. By using `isAssignableFrom` instead of `equals`, we just have a little more resilience to those changes in the future.

Reply:
(insert Wayne's World 'We're Not Worthy' gif here).  Excellent insights as usual, Mike.  Thank you!

Fingers crossed that approach works from the ContextClassLoader workaround we have to do for Databricks.